src/cpusleep.c

```diff
@@ -69,12 +69,9 @@ static int cpusleep_read(void)
 		return (-1);
 	}
 
-	double db = b.tv_sec + 1e-9 * b.tv_nsec;
-	double dm = m.tv_sec + 1e-9 * m.tv_nsec;
-
 	// to avoid false positives in counter overflow due to reboot,
-	// derive is used
-	derive_t sleep = (derive_t) ((db-dm) * 1000);	 
+	// derive is used. Sleep is calculated in milliseconds
+	derive_t sleep = (derive_t) ((b.tv_sec - m.tv_sec)*1e3 + (b.tv_nsec - m.tv_nsec)*1e-6);	 
```

Comment:
It's non-obvious, but `1e3` and `1e-6` are of type `double`, so floating point conversion is still happening.

I'd suggest to use integer arithmetic instead:

```c
derive_t sleep = ((derive_t) (b.tv_sec - m.tv_sec)) * 1000
    + ((derive_t) (b.tv_nsec - m.tv_nsec)) / 1000000;
```